app/frontend/src/features/profile/HostingPreferenceForm.tsx

```diff
@@ -1,10 +1,234 @@
-import { makeStyles } from "@material-ui/core";
-import React from "react";
+import {
+  Checkbox,
+  FormControl,
+  FormControlLabel,
+  makeStyles,
+  Typography,
+} from "@material-ui/core";
+import { Autocomplete } from "@material-ui/lab";
+import { unwrapResult } from "@reduxjs/toolkit";
+import React, { useState } from "react";
+import { Controller, useForm, UseFormMethods } from "react-hook-form";
+import { updateHostingPreference } from "./index";
+import Alert from "../../components/Alert";
+import Button from "../../components/Button";
+import ProfileTextInput from "./ProfileTextInput";
+import { ProfileFormData } from "../../service/user";
+import { useAppDispatch, useTypedSelector } from "../../store";
+import { theme } from "../../theme";
+import { SmokingLocation } from "../../pb/api_pb";
+import { smokingLocationLabels } from "../../constants";
+
+interface HostingPreferenceCheckboxProps {
+  className: string;
+  defaultValue: boolean;
+  name: string;
+  label: string;
+  register: UseFormMethods<ProfileFormData>["register"];
+}
+
+function HostingPreferenceCheckbox({
+  className,
+  defaultValue,
+  label,
+  name,
+  register,
+}: HostingPreferenceCheckboxProps) {
+  return (
+    <FormControl className={className} margin="dense">
+      <FormControlLabel
+        control={<Checkbox color="primary" defaultChecked={defaultValue} />}
+        label={label}
+        name={name}
+        inputRef={register}
+      />
+    </FormControl>
+  );
+}
 
 const useStyles = makeStyles({
-  root: {},
+  alert: {
+    marginBottom: theme.spacing(3),
+  },
+  form: {
+    "& h2": {
```

Comment:
Don't we better avoid referring to html tags?

Reply:
I think these kind of selectors are fine since this is scoped/namespaced to this component only. Although it seems unneeded and was probably left over from me switching over to using Typography. Will remove it!